shared/src/reducers/fadersReducer.ts

```diff
@@ -280,41 +280,42 @@ export const faders = (
             })
             return nextState
         case FADER_ACTIONS.SET_ASSIGNED_CHANNEL:
+            let newAssignment: IChannelReference[] =
+                nextState[0].fader[action.faderIndex].assignedChannels || []
+
             if (action.assigned) {
-                let assignedChannels: IChannelReference[] =
-                    nextState[0].fader[action.faderIndex].assignedChannels || []
-                assignedChannels.push({
-                    mixerIndex: action.mixerIndex,
-                    channelIndex: action.channelIndex,
-                })
-                assignedChannels.sort(
-                    (n1: IChannelReference, n2: IChannelReference) =>
-                        n1.channelIndex - n2.channelIndex
-                )
-                nextState[0].fader[
-                    action.faderIndex
-                ].assignedChannels = assignedChannels
-            } else {
-                let assignedChannels =
-                    nextState[0].fader[action.faderIndex].assignedChannels
                 if (
-                    assignedChannels.find(
-                        (channelReference: IChannelReference) => {
-                            return (
-                                channelReference.channelIndex ===
-                                action.channelIndex
-                            )
-                        }
-                    )
+                    !newAssignment.includes({
+                        mixerIndex: action.mixerIndex,
+                        channelIndex: action.channelIndex,
+                    })
                 ) {
-                    assignedChannels = assignedChannels.filter((channel) => {
-                        return channel !== action.channelIndex
+                    newAssignment.push({
+                        mixerIndex: action.mixerIndex,
+                        channelIndex: action.channelIndex,
                     })
+                    // sort by channelIndex:

```

Comment:
As a rule of thumb: writing comments that explains _what_ is happening is a code smell and useally an indication that something isn't clear enough. (Comments that explains _why_ it happens are fine.)

In this case you could, if you want to, write comparator functions that has the names `channelIndexComparator` and `mixerIndexComparator` or something similar:
```
newAssignment.sort(channelIndexComparator)
newAssignment.sort(mixerIndexComparator)

...

function channelIndexComparator(a: { channelIndex: number }, b: { channelIndex: number }): number {
  return a.channelIndex - b.channelIndex
}

function mixerIndexComparator(a: { mixerIndex: number }, b: { mixerIndex: number }): number {
  return a.mixerIndex - b.mixerIndex
}
```
Now you convey the same message through `newAssignment.sort(channelIndexComparator)` as you do with your comment and you no longer have a comment that you need to remember to maintain.